### PR TITLE
Fix dot metrics issue caused by long getValidDataSizeByLogSegment() call time

### PR DESCRIPTION
**Pull request by zzmao**
Fix dot metrics issue caused by long getValidDataSizeByLogSegment() call time

### CODE REVIEW

**Review by SophieGuo410 — APPROVED** [review]
lgtm！

File: ambry-store/src/main/java/com/github/ambry/store/BlobStoreStats.java [hunk 1]
```diff
@@ -163,7 +164,9 @@ static StatsSnapshot convertStoreUsageToContainerStatsSnapshot(Map<Short, Map<Sh
 
   @Override
   public Pair<Long, Long> getValidSize(TimeRange timeRange) throws StoreException {
+    long start = SystemTime.getInstance().milliseconds();
     Pair<Long, NavigableMap<LogSegmentName, Long>> logSegmentValidSizeResult = getValidDataSizeByLogSegment(timeRange);
+    logger.debug("Time to getValidDataSizeByLogSegment: {} ms", SystemTime.getInstance().milliseconds() - start);
```
Comment (jsjtzyy) [inline ambry-store/src/main/java/com/github/ambry/store/BlobStoreStats.java:169]: Include `storeId` in the debug message

File: ambry-store/src/main/java/com/github/ambry/store/BlobStore.java [hunk 2]
```diff
@@ -260,7 +260,9 @@ public void start() throws StoreException {
               config.storeEnableBucketForLogSegmentReports, time, longLivedTaskScheduler, taskScheduler,
               diskIOScheduler, metrics);
         }
-        metrics.initializeIndexGauges(storeId, index, capacityInBytes, blobStoreStats);
+        if (config.storeEnablePerStoreGauges) {
+          metrics.initializeIndexGauges(storeId, index, capacityInBytes, blobStoreStats);
+        }
```
Comment (jsjtzyy) [inline ambry-store/src/main/java/com/github/ambry/store/BlobStore.java:265]: The config here is fine but I would consider minimizing its scope. I feel like you can move config into `initializeIndexGauges()` method and only control metric registration of `Gauge<Long> currentInvalidDataSize`.

**Review by jsjtzyy — COMMENTED** [review]
lgtm after comments are addressed.

**Review by jsjtzyy — APPROVED** [review]

**Pull request merged**